src/CommonDBTM.php

```diff
@@ -5912,4 +5912,118 @@ public function getMassiveActionsForItem(): MassiveAction
 
         return new MassiveAction($params, [], 'initial', true);
     }
+
+    /**
+     * Automatically update 1-N links tables for the current item.
+     *
+     * @param string $field
+     * @param string $itemtype
+     * @param string $table
+     * @param array  $extra_input
+     *
+     * @return void
+     */
+    protected function update1NTableData(
+        string $field,
+        string $itemtype,
+        string $table,
+        array $extra_input
+    ): void {
+        global $DB;
+
+        // Compute item_1 params (current object)
+        $item_1_fk = getForeignKeyFieldForItemType(static::class);
+        $item_1_id = $this->getID();
+
+        // Compute item_2 params (linked objets(s))
+        $item_2_fk = getForeignKeyFieldForItemType($itemtype);
+
+        // Get input value
+        $input_value = $this->input[$field] ?? null;
+
+        // See dropdownField twig macro, needed for empty values as an empty
+        // array wont be sent in the HTML form
+        $input_defined_value = $this->input["_{$field}_defined"] ?? null;
+
+        // Case 1: no updates -> do nothing
+        if ($input_value === null && !$input_defined_value) {
+            return;
+        }
+
+        // Case 2: input was emptied -> remove all values
+        if (
+            ($input_value === null && $input_defined_value)
+            || (is_array($input_value) && ! count($input_value))
+        ) {
+            $DB->delete($table, $extra_input);
+            return;
+        }
+
+        // Case 3: input was maybe modified -> delete missing values and add new ones
+        $DB->delete($table, array_merge($extra_input, [
+            'NOT'      => [$item_2_fk => $input_value],
+            $item_1_fk => $item_1_id,
+        ]));
+
+        // Get existing values
+        $item_2_ids_db = iterator_to_array($DB->request([
+            'SELECT' => [$item_2_fk],
+            'FROM'   => [$table],
+            'WHERE'  => array_merge($extra_input, [
+                $item_1_fk => $item_1_id,
+            ])
+        ]));
+        $item_2_ids_db = array_column($item_2_ids_db, $item_2_fk);
+
+        // Add new values
+        foreach ($input_value as $item_2_id) {
+            if (in_array($item_2_id, $item_2_ids_db)) {
+                // Value exist
+                continue;
+            }
+
+            $DB->insert($table, array_merge($extra_input, [
+                $item_1_fk => $item_1_id,
+                $item_2_fk => $item_2_id
+            ]));
```

Comment:
Even `CommonDBConnexity` items may have pre/post add operations defined in their class, so this should be done using `CommonDBTM::add()` method.